src/View/Parsers/FilterInterface.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace SilverStripe\View\Parsers;
+
+/**
+ * A FilterInterface is given an input string and returns a filtered string. Replacements will be provided and
+ * performed (typically in regex format), and transliteration may be used as a separate service to replace
+ * characters rather than remove them.
+ *
+ * For example implementations, see {@link URLSegmentFilter}.
+ */
+interface FilterInterface
+{
+    /**
+     * Performs a set of replacement rules against the input string, applying transliteration if a service is
+     * provided, and returns the filtered result.
+     *
+     * @param string $input
+     * @return string
+     */
+    public function filter($input);
```

Comment:
Is this the only method the interface needs?

Reply:
We could add `transliterate` maybe

Reply:
Wait, I guess you're suggesting we remove the getters and setters from the interface?

Reply:
Yes; The bigger an interface the less useful it is. ;) I would only have this method.

Reply:
Yeah true - thanks, I've updated the PR